packages/kit/test/utils.js

```diff
@@ -261,7 +271,7 @@ export const config = {
 	use: {
 		...test_browser_device,
 		screenshot: 'only-on-failure',
-		trace: 'retain-on-failure'
+		trace: 'on-first-retry'
```

Comment:
`PW_DISABLE_TS_ESM` seems like a good idea if it speeds things up, but I'm a bit less excited about this one because it will make flaky tests a lot harder to debug as we'll only get traces if they fail twice in a row. any idea how much of the speed difference if from `PW_DISABLE_TS_ESM` vs this?

Reply:
When just running tests from single packages, disabling tracing is about 4x more significant than the loader. I have to check how it adds up as a whole.

I thought people could reenable tracing through the CLI flag when debugging flaky tests but thinking again, that's near impossible without going into the test directories and flakiness might not repro locally. Maybe it's not worth it, yeah.

Reply:
Hmm, I just noticed that the condition for uploading traces is action failure, it doesn't upload them for flaky tests that pass on a retry, so it doesn't really change anything.

Reply:
> I just noticed that the condition for uploading traces is action failure

It might be a good idea to change that and upload them whenever they exist

> it doesn't upload them for flaky tests that pass on a retry, so it doesn't really change anything

I think it'd still change it a fair amount. E.g. if you have one test that fails three times in a row and ten tests that fail once and then pass, you'd only get the trace for the one test